controllers/update_status.go

```diff
@@ -56,7 +56,6 @@ func (updateStatus) reconcile(ctx context.Context, r *FoundationDBClusterReconci
 	// Pass through Maintenance Mode Info as the maintenance_mode_checker reconciler takes care of updating it
 	originalStatus.MaintenanceModeInfo.DeepCopyInto(&clusterStatus.MaintenanceModeInfo)
 	clusterStatus.Generations.Reconciled = cluster.Status.Generations.Reconciled
-	clusterStatus.ConnectionString = originalStatus.ConnectionString
```

Comment:
I am not sure even this change is entirely correct. This line is essentially copying the connection string from "originalStatus" to "clusterStatus". Instead you are going to update "clusterStatus.connectionString" in "getStatusFromClusterOrDummyStatus()" now. But what if "clusterStatus.connectionString" won't get updated in that function (because "tryConnectionOptions()" returns an empty connection string)?

Reply:
Actually, it appears you had added this line in your previous commit?

Reply:
But, overall, I like the idea of updating connectionString in a single place, rather than spreading it across multiple functions.

Reply:
Correct, the was from an old commit that got removed.